github/actions_workflow_runs.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2020 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// PersonWorkflowRun represents a person object as returned in actions workflow jobs endpoint response.
+type PersonWorkflowRun struct {
+	Name  *string `json:"name,omitempty"`
+	Email *string `json:"email,omitempty"`
+}
+
+// CommitWorkflowRun represents a commit object as returned in actions workflow jobs endpoint response.
+type CommitWorkflowRun struct {
+	ID        *string            `json:"id,omitempty"`
+	TreeID    *string            `json:"tree_id,omitempty"`
+	Message   *string            `json:"message,omitempty"`
+	Timestamp *Timestamp         `json:"timestamp,omitempty"`
+	Author    *PersonWorkflowRun `json:"author,omitempty"`
+	Committer *PersonWorkflowRun `json:"committer,omitempty"`
+}
+
+// WorkflowRun represents a repository action workflow run.
+type WorkflowRun struct {
+	ID             *int64            `json:"id,omitempty"`
+	NodeID         *string           `json:"node_id,omitempty"`
+	HeadBranch     *string           `json:"head_branch,omitempty"`
+	HeadSHA        *string           `json:"head_sha,omitempty"`
+	RunNumber      *int64            `json:"run_number,omitempty"`
+	Event          *string           `json:"event,omitempty"`
+	Status         *string           `json:"status,omitempty"`
+	Conclusion     *string           `json:"conclusion,omitempty"`
+	URL            *string           `json:"url,omitempty"`
+	HTMLURL        *string           `json:"html_url,omitempty"`
+	PullRequests   []*PullRequest    `json:"pull_requests,omitempty"`
+	CreatedAt      *Timestamp        `json:"created_at,omitempty"`
+	UpdatedAt      *Timestamp        `json:"updated_at,omitempty"`
+	JobsURL        *string           `json:"jobs_url,omitempty"`
+	LogsURL        *string           `json:"logs_url,omitempty"`
+	CheckSuiteURL  *string           `json:"check_suite_url,omitempty"`
+	ArtifactsURL   *string           `json:"artifacts_url,omitempty"`
+	CancelURL      *string           `json:"cancel_url,omitempty"`
+	RerunURL       *string           `json:"rerun_url,omitempty"`
+	HeadCommit     CommitWorkflowRun `json:"head_commit,omitempty"`
```

Comment:
I see in this repo that we have a very similar struct that appears to serve the same purpose that I think might be good to reuse here: `PushEventCommit` in `event_types.go`, although some might argue with me that `PushEventCommit` is strictly to be used for events.

I would argue that the `HeadCommit` is how that struct is being used, and maybe it should be renamed to `HeadCommit` and then used in both places to represent the same thing.

Either way, whether `PushEventCommit` is renamed to `HeadCommit` in this PR or not, this instance of `HeadCommit` should be a pointer to a struct, not an embedded struct.

As for the rename of `PushEventCommit`, I'll leave it up to you and the second reviewer of this PR, as I could go either way.